firestore-send-email/functions/package.json

```diff
@@ -8,7 +8,10 @@
     "build:watch": "npm run clean && tsc --watch",
     "clean": "rimraf lib",
     "compile": "tsc",
-    "test": "jest",
+    "local:emulator": "cd ../../_emulator && firebase emulators:start -P demo-test",
+    "test": "cd ../../_emulator && firebase emulators:exec jest -P demo-test",
+    "test:local": "concurrently \"npm run local:emulator\" \"jest\"",
```

Comment:
Is there any potential for a race condition here? ie can the tests start running before the emulator starts up?

Reply:
Yes, there could be. I'm currently refining in the background, we could for example run a `sh` script, something like....

```bash
#!/bin/bash
if ! [ -x "$(command -v firebase)" ]; then
  echo "❌ Firebase tools CLI is missing."
  exit 1
fi

until curl --output /dev/null --silent --fail http://localhost:9099; do
  echo "Waiting for Firestore emulator to come online..."
  sleep 2
done

echo "Firestore emulator is online!" 

cd ../functions
jest --watch

```